index.js

```diff
@@ -1,16 +1,36 @@
 'use strict';
 module.exports = function (hex) {
-	if (typeof hex !== 'string') {
-		throw new TypeError('Expected a string');
+	if (
+		typeof hex !== 'string' ||
+		/[^#a-f0-9]/gi.test(hex) ||
+		!/^#?[a-f0-9]{3}[a-f0-9]?$|^#?[a-f0-9]{6}([a-f0-9]{2})?$/i.test(hex)
+	) {
+		throw new TypeError('Expected a valid hex string');
 	}
 
 	hex = hex.replace(/^#/, '');
+	let alpha = 255;
+
+	if (hex.length === 8) {
+		alpha = parseInt(hex.substring(6, 8), 16);
```

Comment:
Use `slice` instead of `substring`. See: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/65